backend-schedule/src/main/java/eu/socialedge/hermes/backend/schedule/domain/BasicScheduleGenerator.java

```diff
@@ -0,0 +1,183 @@
+            nextTimePoint = nextTimePoint.plus(headway);
+        }
+        return timePoints;
+    }
+
+    private List<Trip> generateVehicleTrips(int vehicleId, TimePoint startPoint, List<TimePoint> timePoints) {
+        val trips = new ArrayList<Trip>();
+
+        while (true) {
+            val trip = generateTrip(vehicleId, startPoint);
+            trips.add(trip);
+
+            //TODO maybe remove last trip and break if its arrival time is after end time? May be some parameter to indicate possible lateness?
+            val currentTime = getArrivalTime(trip);
+            val currentDirection = INBOUND.equals(startPoint.direction()) ? OUTBOUND : INBOUND;
+            val nextPointOpt = findNextNotServicedTimePointAfter(timePoints, currentTime, currentDirection);
+
+            if (nextPointOpt.isPresent()) {
+                startPoint = nextPointOpt.get();
+            } else {
+                break;
+            }
+        }
+
+        return trips;
+    }
+
+    private Trip generateTrip(int vehicleId, TimePoint timePoint) {
+        timePoint.isServiced(true);
+        val route = INBOUND.equals(timePoint.direction()) ? routeInbound : routeOutbound;
+        return new Trip(
+            route,
+            vehicleId,
+            route.stations().get(route.stations().size() - 1).name(),
+            calculateStops(timePoint.time(), route, averageSpeed, dwellTime));
+    }
+
+    private List<Stop> calculateStops(LocalTime startTime, Route route, Quantity<Speed> averageSpeed, Duration dwellTime) {
+        val stops = new ArrayList<Stop>();
+
+        val speedValue = averageSpeed.to(Units.METRE_PER_SECOND).getValue().longValue();
+
+        for (int i = 0; i < route.stations().size(); i++) {
+            val station = route.stations().get(i);
+            val distTravelled = route.shape().shapePoints().stream()
+                .filter(shapePoint -> station.location().equals(shapePoint.location()))
+                .findFirst()
+                .orElseThrow(() -> new ScheduleGeneratorException("Could not match stations with route shape. Station not found with location + " + station.location()))
+                .distanceTraveled();
+
+            val distValue = distTravelled.to(Units.METRE).getValue().longValue();
+            val arrivalTime = startTime.plusSeconds(distValue / speedValue).plus(dwellTime.multipliedBy(i));
+
+            stops.add(new Stop(arrivalTime, arrivalTime.plusSeconds(dwellTime.getSeconds()), station));
+        }
+
+        return stops;
+    }
+
+    private boolean isInTimeToTravel(LocalTime from, TimePoint toPoint) {
+        return !Duration.between(from, toPoint.time()).minus(minLayover).isNegative();
+    }
+
+    private static boolean hasNotServicedTimePoints(List<TimePoint> timePoints) {
+        return timePoints.stream().anyMatch(timePoint -> !timePoint.isServiced());
+    }
+
+    private static TimePoint getNextNotServicedTimePoint(List<TimePoint> timePoints) {
```

Comment:
I understand "find" like looking for something that might not exist. But this method assumes that next not services time point exists, so just "gets" it